spec/factories/profile_link_sites.rb

```diff
@@ -13,5 +13,123 @@
 FactoryGirl.define do
   factory :profile_link_site do
     name { Faker::Company.name }
+    validate_find 'use trait'
+    validate_replace 'use trait'
+
+    # rubocop:disable Metrics/LineLength
+    # Twitter
+    trait :twitter do
+      validate_find '(https://)?(www.)?(twitter.com/)?(@)?(?<username>[a-zA-Z0-9_]+)'
+      validate_replace 'https://twitter.com/\k<username>'
+    end
+
+    # Facebook
+    trait :facebook do
+      validate_find '(https://)?(www.)?(facebook.com/)?(?<username>[a-zA-z0-9_.]+)'
+      validate_replace 'https://facebook.com/\k<username>'
+    end
+
+    # Youtube
+    trait :youtube do
+      validate_find '(https://)?(www.)?(youtube.com/channel/)?(?<username>[a-zA-z0-9_\-]+)'
+      validate_replace 'https://youtube.com/channel/\k<username>'
+    end
+
+    # Google
+    trait :google do
+      validate_find '(https://)?(www.)?(plus.google.com/)?(?<username>.+)'
+      validate_replace 'https://plus.google.com/\k<username>'
+    end
+
+    # Instagram
+    trait :instagram do
+      validate_find '(https://)?(www.)?(instagram.com/)?(?<username>[a-zA-z0-9_.]+)'
+      validate_replace 'https://instagram.com/\k<username>'
+    end
+
+    # Twitch
+    trait :twitch do
+      validate_find '(https://)?(www.)?(twitch.tv/)?(#)?(?<username>[a-zA-z0-9_]+)'
+      validate_replace 'https://twitch.tv/\k<username>'
+    end
+
+    # Vimeo
+    trait :vimeo do
+      validate_find '(https://)?(www.)?(vimeo.com/)?(?<username>[a-zA-z0-9_\-]+)'
+      validate_replace 'https://vimeo.com/\k<username>'
+    end
+
+    # Github
+    trait :github do
+      validate_find '(https://)?(www.)?(github.com/)?@?(?<username>[a-zA-z0-9\-]+)'
+      validate_replace 'https://github.com/\k<username>'
+    end
+
+    # Battlenet
+    trait :battlenet do
+      validate_find '(.+#[0-9]+)'
+      validate_replace '\1'
+    end
+
+    # Steam
+    trait :steam do
+      validate_find '((http|https)://)?(www.)?(steamcommunity.com/id/)?(?<username>[a-zA-z0-9_\-]+)'
```

Comment:
Neither would achieve the desired aim.

The `?` must _directly_ modify the `s`, because `?` means "previous character/grouping, 0 or 1 times"